### ISSUE TITLE
question: How can the parent properties be reached in the nested object?

### ISSUE THREAD

**Issue by eomelyusik** [typestack/class-validator#1469]
I was trying to validate the nested object and faced an issue when it is not possible to reach the parent properties in the nested object:

Parent class:
```
export class ParentDto {
  @Type(() => NestedDto)
  @ValidateNested()
   nestedProperty: NestedDto;
   type: string;
}
```

Nested class:

```
export class NestedDto {
  @ValidateIf((object) => object.type === 'some-type')
  @IsString()
  @IsNotEmpty()
  value: string;
}
```

Is it possible to reach somehow "type" property from the ParentDto in the NestedDto?
The only solution I get is to use @Transform to pass type in the NestedDto.

**Comment by adityajadhav95** [comment]
Is there any workaround found for this problem?

**Comment by glothos** [comment]
I need this as well. Trying out some custom validators.

**Comment by IrfanHasib** [comment]
I'm struggling to find similar feature.

**Comment by zhfahan77** [comment]
Same here, is there any workaround or any actual way to do this?

**Comment by Iuriiiiiii** [comment]
Has anyone an answer for this? :(

**Comment by aaleksandrov** [comment]
You have to use Custom validator, in [the example](https://github.com/typestack/class-validator?tab=readme-ov-file#custom-validation-decorators)  `args.object` is the entire object

**Comment by Ma-An** [comment]
@aaleksandrov I've tried this option and it looks like the `args.object` only has access to the object being validated and not the root instance properties.

**Comment by aaleksandrov** [comment]
@Ma-An No, in my team we use this feature extensively when we need access to the root object. I am a bit lazy to provide a working example though, sorry. But it's working like this
UPD: you mean the very root object? You are right then
